lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,551 @@
+    returned: always
+    type: dict
+    sample: {
+        "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing",
+        "location": "westus",
+        "name": "Testing",
+        "provisioning_state": "Succeeded",
+        "tags": {
+            "delete": "on-exit",
+            "testing": "no"
+        }
+    }
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.trafficmanager.models import DnsConfig, Profile, MonitorConfig, Endpoint
+except ImportError:
+    pass
+
+
+def trafficmanagerprofile_group_to_dict(tm):
+    return dict(
+        id=tm.id,
+        name=tm.name,
+        resource_group=tm.resource_group,
+        location=tm.location,
+        tags=tm.tags)
+
+
+class AzureRMTrafficManager(AzureRMModuleBase):
+    # Main Traffic Manager class
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            name=dict(type='str', required=True),
+            resource_group=dict(type='str', required=True),
+            properties=dict(type='dict'),
+            location=dict(type='str'),
+            status=dict(type='str', default='Enabled', choices=['Enabled', 'Disabled']),
+            state=dict(type='str', default='present', choices=['present', 'absent'])
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.properties = None
+        self.location = None
+        self.status = None
+        self.state = None
+
+        self.results = dict(
+            changed=False,
+            contains_endpoints=False,
+            state=dict())
+
+        super(AzureRMTrafficManager, self).__init__(self.module_arg_spec,
+                                                    supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        # Collect all the tags and add them to the attributes
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        # Before we proceed, verify that the resource group exists.
+        try:
+            self.get_resource_group(self.resource_group)
+
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
+
+        # Initialize the Ansible return
+        results = dict()
+        changed = False
+        current_traffic_manager = self.get_traffic_manager_profile(
+            self.resource_group,
+            self.name)
+
+        results = self.traffic_manager_to_dict(
+            current_traffic_manager,
+            self.resource_group,
+            self.name, self.location,
+            self.properties)
+
+        try:
+            if self.state == 'present':
+                self.log('Fetching traffic manager {0}'.format(self.name))
+                # dry-run.
+                if self.check_mode:
```

Comment:
Why not check the `self.check_mode` condition before call the create/upate/delete API, this can reduce the repeated work in each conditional branch